### ISSUE TITLE
Keybase won't start until network mounts in my login items are mounted

### ISSUE THREAD

**Issue by geofffranks** [keybase/client#6359]
Trying to launch the new Keybase GUI app on macOS Sierra (for the first time). I have an AFP fileshare in my login items (to connect to my backup share on my NAS). When not connected to my network, Keybase hangs and I eventually see finder errors about being unable to connect to my NAS. When connected to my home network, it immediately prompts me a multitude of times to connect to my NAS, before Keybase asks me for my password to instal helper utilities.

If I remove the NAS login item from my user, It launches immediately, and prompts me for my password to install the helper utilities. 

Why does Launching keybase attempt to open up all my login items repeatedly, and connect to my remote servers? While this may not be intentional or malicious, Keybase is security software that shouldn't know about the existence my NAS. When it requires me to log into something that it shouldn't even know about in order to start up, that puts a really bad taste in my mouth and raises a lot of questions. How did it find out about the NAS? What is it trying to do with the data on it? What other personal data that it has no business being in is it looking at?

**Comment by strib** [comment]
cc: @gabriel 

I don't know much about the macOS startup stuff, hopefully @gabriel can answer, but I can promise you Keybase is not trying to inspect any of your local data.  Perhaps one of the subdirectories in your home directory where we store data isn't available until that drive either gets mounted or errors out?  Or maybe our file system driver (osxfuse) hangs while waiting for the mount to finish?

Sorry I can't help much more than speculating wildly.  If you'd like us to look more in depth, you can do a `keybase log send` and let us know the resulting ID, and we can try to figure out why it's hanging.

**Comment by geofffranks** [comment]
Since this is occurring before the Keybase app can start + install things, I don't have a `keybase log send` command. As far as I'm aware, the only place the mount is referenced is in the login item preferences. When I remove it via `osascript -e 'tell application "System Events" to delete login item "Backup"'`, Keybase no longer hangs. I'll grab a fresh copy of ~/Library/Logs/Keybase.app.log tonight when I get home, if that helps.

**Comment by gabriel** [comment]
Yeah I think this is our bug. We look over Finder favorites to see if we need to Keybase to the list.
It's possible it's querying finder items without setting a flag to not trigger a mount. (It's not accessing anything on the share, it's just accidentally triggering a mount.)

I'll post more details and the fix here in the next few days.

Thanks for the bug report. Sorry you hit this issue.

**Comment by kglick** [comment]
So, any comments one when this may be resolved.  Like Geoff I'm bothered by the app wanting to mount/access my login shares.  Unlike Geoff if I click cancel a dozen or so times it stops asking and launches the app.  I'm running OSX 10.11.6

**Comment by kglick** [comment]
Folks.  Do we have an update on this bug?

**Comment by strib** [comment]
I will file an internal issue about this, if we have time we'll look into it.  As gabriel said above, we're not accessing the shares themselves, which you can verify in our open source code, but this is reported so rarely that it hasn't been a priority to fix.

**Comment by strib** [comment]
(Also note you can disable the Keybase mount from the files tab of the Keybase GUI, if you don't want it on.)

**Comment by kglick** [comment]
Jeremy,

 

Ive uncheck that box and relaunched the app.  Unfortunately, Im still having the same issue.  Id be happy to send you the current keybase.system.log.  To where shall I send it?

 

-k

**Comment by strib** [comment]
You can run `keybase log send` and post the resulting log ID here.

Can you clarify exactly which box you unchecked though?

**Comment by kglick** [comment]
See screenshot.  I disabled keybase in finder.  Settings>uncheck enable keybase in finder.  I believe when its check this box actually shows up on the files tab.  Logs on the way

 

-k

**Comment by strib** [comment]
Thanks.  Logs: 7f60fab56b5cb06c49cc031c

**Comment by andresv** [comment]
I have also problem with network mounts - everytime I activate keybase window, network mount error pops up.
![Screenshot 2019-04-05 22 06](https://user-images.githubusercontent.com/809232/55651047-0dfd9180-57f0-11e9-80e2-b71f36deb4f6.jpg)
Keybase should not be interested in this drive. I also disabled Finder file integration, but the problem is still there.

**Comment by kglick** [comment]
Yes, this has been going on for the better part of three years now.  It would be greatly appreciated if it got fixed.

Kev

**Comment by andresv** [comment]
What kind of information you would like to have to get this fixed. It is extremely annoying and keybase is totally unusable for my team because of that.

So let me know if there is anything I can provide.
